learningequality/studio
# Replace PrimaryDialog with KModal (and remove it along with DialogBox)

## Description

- [ ] Replaces all uses of `PrimaryDialog` with `KModal`
- [ ] Removes `PrimaryDialog` and `DialogBox` from the codebase (the latter was not used to begin with)
- [ ] Applies changes to other components to ensure proper functionality.

#### Issue Addressed (if applicable)

Fixes https://github.com/learningequality/kolibri-design-system/issues/158


## Steps to Test

- Copy the channel token from the Channel List
- Copy the channel token from the Channel Edit
- Use the Channel Edit -> Sync Resources Modal - Ensure that you can go back and forth between the steps.
- See this comment for testing the Catalog Filters Bar: https://github.com/learningequality/studio/commit/8f281cc1a697d0ed2a57911bdd7160b3ffedb426#r47435467

## Implementation Notes (optional)

#### At a high level, how did you implement this?

`KModal` doesn't take a v-model, so we change those to v-ifs.
Then instead of using the actions slot (that both KModal and VDialog have) - we use the simpler props on KModal to apply the text for the cancel/submit buttons and the actions to take when they're clicked.

#### Does this introduce any tech-debt items?

Quite the opposite :)
